src/renderers/webxr/WebXRManager.js

```diff
@@ -609,6 +615,120 @@ class WebXRManager extends EventDispatcher {
 
 		};
 
+		this.getPlaneType = function ( plane ) {
```

Comment:
Would it be reasonable to estimate plane types using a `bounded-floor` reference space, and (at least for now) omit plane type detection in other reference spaces? I'm hoping that could be cheaper and simpler, as each plane can be simply compared against the reference space, without needing to iterate all vertices of all other planes. This also removes timing-dependency on when a plane is detected, and its type estimated, relative to other planes.

Not to say I don't want to try to support plane detection in other modes, when possible.

Given that plane dimensions are in meters, we could also make a reasonable estimate based on size and orientation, without necessarily comparing planes against one another.

If there's no clear approach that's both robust and fast, this might also be something we want to put into a utility class (`examples/jsm/webxr/XRPlaneUtils.js`?) to leave more room for exploring other approaches.

Reply:
Sounds good, I can make it into an example, tonight :)